lib/matplotlib/backends/backend_agg.py

```diff
@@ -208,10 +208,13 @@ def draw_text(self, gc, x, y, s, prop, angle, ismath=False, mtext=None):
         xo, yo = font.get_bitmap_offset()
         xo /= 64.0
         yo /= 64.0
+        h = sqrt(xo**2 + yo**2)
         xd = d * sin(radians(angle))
         yd = d * cos(radians(angle))
+        xo = h * cos(radians(angle))
+        yo = h * sin(radians(angle))
```

Comment:
Also, these four lines here would greatly benefit from a comment explaining the trigonometry. Usually, cos() goes with x, and y goes with sin(). Seeing it two different ways here for the offset and the descent is confusing.

Reply:
I can explain the offset trigonometry, but I just left the code for the descent as it was before. The descent seems wrong - but I am unaware of the reason why it was like that before so I left it there. Basic trigonometry suggests that x components should be calculated by multiplying a number by the cosine of its angle and y components should be calculated by multiplying a number by the sine of its angle - so that is why I used that for the offset but I do think the descent should be in the same form too.